docs/tuning-apps/mpi-io/ompio.rst

```diff
@@ -0,0 +1,262 @@
+
+   shell$ ompi_info --param io       ompio --level 9
+   shell$ ompi_info --param fcoll    all --level 9
+   shell$ ompi_info --param fs       all --level 9
+   shell$ ompi_info --param fbtl     all --level 9
+   shell$ ompi_info --param sharedfp all --level 9
+
+OMPIO sub-framework components
+------------------------------
+
+The OMPIO architecture is designed around sub-frameworks, which allow
+you to develop a relatively small amount of code optimized for a
+particular environment, application, or infrastructure.  Although
+significant efforts have been invested into making good decisions for
+default values and switching points between components, users and/or
+system administrators might occasionally want to tune the selection
+logic of the components and force the utilization of a particular
+component.
+
+The simplest way to force the usage of a component is to simply
+restrict the list of available components for that framework. For
+example, an application wanting to use the ``dynamic`` ``fcoll``
+component simply has to pass the name of the component as a value to
+the corresponding MCA parameter during ``mpirun`` or any other
+mechanism available in Open MPI to influence a parameter value, e.g.:
+
+.. code-block:: sh
+
+   shell$ mpirun --mca fcoll dynamic -n 64 ./a.out
+
+``fs`` and ``fbtl`` components are typically chosen based on the file
+system type utilized (e.g. the ``pvfs2`` component is chosen when the
+file is located on an PVFS2/OrangeFS file system, the ``lustre``
+component is chosen for Lustre file systems, etc.). The ``ufs`` ``fs``
+component is used if no file system specific component is availabe
+(e.g. local file systems, NFS, BeefFS, etc.), and the ``posix``
+``fbtl`` component is used as the default component for read/write
+operations.
+
+The ``fcoll`` framework provides several different components. The
+current decision logic in OMPIO uses the file view provided by the
+application as well as file system level characteristics (e.g. file
+system, stripe width) to determine which component to use. The most
+important ``fcoll`` components are:
+
+* ``dynamic_gen2``: the default component used on lustre file
+  system. This component is based on the two-phase I/O algorithm with
+  a static file partioning strategy, i.e. an aggregator processes will
+  by default only write data to a single storage server.
+
+* ``vulcan``: the default component used on all other file
+  systems. This component is based on the two-phase I/O algorithm with
+  an even file partitioning strategy, i.e. each of the n aggregators
+  will write 1/n th of the overall file.
+
+* ``individual``: this components executes all collective I/O
+  operations in terms of individual I/O operations.
+
+The ``sharedfp`` framework provides a different implementation of the
+shared file pointer operations depending on file system features.
+
+* ``lockfile``: this component will be used on file system which
+  support for file locking.
```

Comment:
re-reading I noticed a small issue here: it should probably be 'this component will be used on file systems which support file locking' (plural for file syste(s) and should remove 'for')